layer/contracts/models.py

```diff
@@ -18,9 +17,36 @@
 
 @dataclass(frozen=True)
 class TrainStorageConfiguration:
-    train_id: ModelTrainId
+    train_id: uuid.UUID
     s3_path: S3Path
-    credentials: AwsCredentials
+    credentials: AWSCredentials
+
+
+@enum.unique
+class ModelTrainStatus(enum.IntEnum):
+    INVALID = 0
+    PENDING = 1
+    INITIALIZING = 2
+    FETCHING_FEATURES = 3
+    IN_PROGRESS = 4
+    SUCCESSFUL = 5
+    FAILED = 6
+    CANCEL_REQUESTED = 7
+    CANCELED = 8
```

Comment:
why does it need to be an IntEnum and not simply an Enum?

Reply:
this is based off of the DatasetBuildStatus for consistency.